### ISSUE TITLE
Suggestion: Save a JSON File while saving the model

### ISSUE THREAD

**Issue by gaotianxiang** [keras-team/keras#8251]
Since we typically train the model at different hyper parameter settings at the beginning, I believe that a file (maybe json) is helpful to be generated at the same time when the model is saved. The file record the core information about the model, such as batch size, learning rate, epoch number, model architecture, and optimizer settings.

Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).
